### DISCUSSION TITLE
Mismatch of body vs query parameter when using dependency injection

### DISCUSSION THREAD

**Discussion by mdespriee** [fastapi/fastapi#8719] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
(see inline)
```

### Description

Hello,
I'm building an app relying on dependency injection and classes for payload models.
My problem is that half of the payload model is interpreted as query params, the other half as body. 
I want the whole object to be passed as body.

The model is declared like this:
```
class HourlyExportRequest(BaseModel):
    metrics: List[Metric]
    devices: List[DeviceWithReplacement]
    name: Optional[str]
    identification: str
```

and here the API endpoint:
```
@router.post("/export/hourly", response_model=ExportResponse)
@inject
async def export_hourly(
    body: HourlyExportRequest = Depends(HourlyExportRequest),
    export_handler: ExportHandler = Depends(Provide[Container.export_handler]),
) -> ExportResponse:
    return await export_handler.export_hourly(body)
```

Here is the generated openAPI  
```
   "/export/hourly": {
      "post": {
        "summary": "Export Hourly",
        "operationId": "export_hourly_export_hourly_post",
        "parameters": [
          {
            "required": false,
            "schema": {
              "title": "Name",
              "type": "string"
            },
            "name": "name",
            "in": "query"
          },
          {
            "required": true,
            "schema": {
              "title": "Identification",
              "type": "string"
            },
            "name": "identification",
            "in": "query"
          }
        ],
        "requestBody": {
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/Body_export_hourly_export_hourly_post"
              }
            }
          },
          "required": true
        },
    ...
```

In the docs, it seems the solution is around Body(), but I struggle to combine it with dependency injection.

### Operating System

Linux

### FastAPI Version

0.70.1

### Python Version

3.9

**Comment by mdespriee** [comment]
Well, maybe I don't need the dependency injection for the model :thinking: 

The following works as expected...
```
@router.post("/export/hourly", response_model=ExportResponse)
@inject
async def export_hourly(
    body: HourlyExportRequest,
    export_handler: ExportHandler = Depends(Provide[Container.export_handler]),
) -> ExportResponse:
    return await export_handler.export_hourly(body)
```

Now, I think I don't understand the value of dependency injection for models.

**Comment by adriangb** [comment, marked as answer]
Dependency injection is for things like a database connection of HTTP client. Not for the JSON body.
